.gitignore

```diff
@@ -1,11 +1,104 @@
-# Global ignores
-__pycache__/
+# Wildcards
+*.8
+*.a
+*.ao
+*.d
+*.dSYM
+*.db
+*.dll
+*.dylib
+*.exp
+*.lib
+*.lnk
+*.ncb
+*.o
+*.obj
+*.opendb
+*.opendf
+*.opensdf
+*.opt
+*.out
+*.pc
+*.pdb
+*.plg
 *.pyc
+*.s
+*.sdf
+*.so
+*.suo
+*.tgz
+*.user
+*.vcproj.*.*.user
+*.vcxproj.user
+*_debug
+*_release
+cygicudata*
+libicu*
+libicutest*
+libsicu*
+libsicutest*
+output-*.cpp
+perf-*.xml
+test-*.xml
+
+# Directories
+.vs/
+.vscode/
+ARM/
+ARM64/
+Debug/
+Generated[!!-~]Files/
+Release/
+__pycache__/
+arm/
+arm64/
+bin/
+debug/
+doc/
+lib/
+out/
+release/
+tmp/
+x64/
+x86/
+
+# Files
+Makefile
+Makefile.local
+build-local.properties
+icupkg.inc
+ipch
+pkgdata.inc
+pkgdataMakefile
+rules.mk
+
+!icu4c/source/samples/csdet/Makefile
+
+# Directory specific wildcards
+icu4c/source/extra/uconv/resources/*.res
+icu4c/source/perf-*.xml
+icu4c/source/samples/udata/*.dat
+icu4c/source/samples/ufortune/resources/*.dat
+icu4c/source/samples/ufortune/resources/*.res
+icu4c/source/samples/uresb/*.res
+icu4c/source/test-*.xml
+icu4c/source/test/hdrtst/ht_*
+icu4c/source/test/perf/howExpensiveIs/*.xml
+icu4c/source/tools/gendict/*.1
+icu4c/source/tools/genrb/*.1
+icu4c/source/tools/genrb/derb_*
+icu4c/source/tools/genren/urename.*
+icu4c/source/tools/makeconv/*.1
+icu4c/source/tools/pkgdata/*.1
+icu4j/*.jar
+icu4j/lib/*.jar
+tools/multi/packages/*.jar
+tools/release/java/APIChangeReport*
 
-# Specific file name ignores
-/.git
-.vscode
-icu4c/bin
+icu4c/source/tools/icuinfo/icuplugins*.txt
+!icu4c/source/tools/icuinfo/icuplugins_windows_sample.txt
+
+# Specific file names
```

Comment:
super-nit: Some of these are really "files", rather they're folders/directories.

Reply:
That's just my Unix background shining through: Every name in the file system is a file name, regardless of whether it refers to a regular file, a directory, a symbolic link, a device, etc., etc. ;-)